torch/nn/modules/transformer.py

```diff
@@ -89,14 +89,15 @@ def forward(self, src, tgt, src_mask=None, tgt_mask=None,
             - tgt_key_padding_mask: :math:`(N, T)`.
             - memory_key_padding_mask: :math:`(N, S)`.
 
-            Note: [src/tgt/memory]_mask should be filled with
-            float('-inf') for the masked positions and float(0.0) else. These masks
-            ensure that predictions for position i depend only on the unmasked positions
-            j and are applied identically for each sequence in a batch.
-            [src/tgt/memory]_key_padding_mask should be a ByteTensor where True values are positions
-            that should be masked with float('-inf') and False values will be unchanged.
-            This mask ensures that no information will be taken from position i if
-            it is masked, and has a separate mask for each sequence in a batch.
+            Note: [src/tgt/memory]_mask ensures that position i is allowed to attend the unmasked
+            positions. If a ByteTensor is provided, positions with ``1`` are not allowed to attend
```

Comment:
For ByteTensor, since it's converted into a BoolTensor, if a position is something other than 1 or 0 (for example 15), do we raise an error or do we get a consistent conversion from say 15 to 1?

If so I'd say "If a ByteTensor is provided **non-zero** positions are not allowed to attend [...]```

Reply:
A question: what's reasoning behind extra-supporting ByteTensors? back-compat? Since a while PyTorch transitioned to BoolTensors and all binary ops return BoolTensors. If the input is supposed to be produced by PyTorch-provided methods like generate_subsequence_mask, PyTorch can return BoolTensor only

Reply:
> A question: what's reasoning behind extra-supporting ByteTensors? back-compat? Since a while PyTorch transitioned to BoolTensors and all binary ops return BoolTensors. If the input is supposed to be produced by PyTorch-provided methods like generate_subsequence_mask, PyTorch can return BoolTensor only

It's because some downstream libraries (like fairseq) use bytetensors for key/attention masks so it's better to support it here.

Reply:
> For ByteTensor, since it's converted into a BoolTensor, if a position is something other than 1 or 0 (for example 15), do we raise an error or do we get a consistent conversion from say 15 to 1?
> 
> If so I'd say "If a ByteTensor is provided **non-zero** positions are not allowed to attend [...]```

Good catch. I will update the doc.